### ISSUE TITLE
ArrayIndexOutOfBoundsException when processing 837 file

### ISSUE THREAD

**Issue by menacher** [imsweb/x12-parser#9]
I get the following error when processing an 837 file. Any idea why?
```java
java.lang.ArrayIndexOutOfBoundsException: -1
	at java.util.ArrayList.elementData(ArrayList.java:418) ~[na:1.8.0_91]
	at java.util.ArrayList.get(ArrayList.java:431) ~[na:1.8.0_91]
	at com.imsweb.x12.Loop.getLoop(Loop.java:285) ~[x12-parser-1.4.jar:1.4]
	at com.imsweb.x12.reader.X12Reader.storeData(X12Reader.java:362) ~[x12-parser-1.4.jar:1.4]
	at com.imsweb.x12.reader.X12Reader.parse(X12Reader.java:214) ~[x12-parser-1.4.jar:1.4]
	at com.imsweb.x12.reader.X12Reader.<init>(X12Reader.java:98) ~[x12-parser-1.4.jar:1.4]
```
Please see the 837 file attached which causes this exception. [st.txt](https://github.com/imsweb/x12-parser/files/2259979/st.txt)

My code:
```java
        URL url = this.getClass().getResource("/st.txt");
        X12Reader reader = null;
        try {
            reader = new X12Reader(X12Reader.FileType.ANSI837_5010_X222, new File(url.getFile()));
            List<String> errors = reader.getErrors();
            if(errors != null && errors.size() > 0) {
                log.error("Parse errors: {}", errors);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
```

From some basic debugging the issue is that index is -1 in the code below:
```java
/**
     * Returns the child loop at the specified index of the loop with the corresponding loopID.
     * @param loopId is the loop we are searching for
     * @param index is the child loop index we want.
     * @return loop with corresponding loopId. Returns null if the index is greater than or equal to the number of loops matching the loop ID.
     */
    public Loop getLoop(String loopId, int index) {
        List<Loop> loops = findLoop(loopId);
        if (index < loops.size())
            return findLoop(loopId).get(index);
        return null;
    }
```

Also, why call findLoop again inside if condition? shouldnt loops.get(index) work the same as it is already processed earlier in the method?

**Comment by angelaszek** [comment]
Thanks for your feedback. I looked at the file you provided. The cause of the error is that the first line of the file is not the expected 106 characters (the ~ needed to be the 106th character in the line). This caused problems getting the separators needed for parsing the file. I improved our checks that we do on the first line of a claim to make sure that a meaningful error message is returned instead of this exception. 

Another thing I noticed about your file is that the claim version is 005010X223A2 and the constructor you are using in your code is for files that have version 005010X222A1. This didn't cause the error but I also added checks to make sure that the file type specified in the constructor of the reader is the same as the version specified in the file.
